File: backend/internal/model/article.go
package model

import "time"

const (
	ArticleSourceTypeManual            = "manual"
	ArticleSourceTypeKnowledgeDocument = "knowledge_document"
)

// Article 文章模型
type Article struct {
	ID          int64  `gorm:"primaryKey;autoIncrement" json:"id"`
	Title       string `gorm:"size:200;not null" json:"title"`
	Slug        string `gorm:"size:200;not null;uniqueIndex" json:"slug"`
	Summary     string `gorm:"type:text" json:"summary"`
	Content     string `gorm:"type:longtext;not null" json:"content"`
	ContentHTML string `gorm:"type:longtext" json:"content_html"` // 渲染后的 HTML 缓存

	CategoryID int64 `gorm:"not null;index:idx_category" json:"category_id"`
	AuthorID   int64 `gorm:"not null" json:"author_id"`

	CoverImage    *string `gorm:"size:500" json:"cover_image,omitempty"`
	Status        string  `gorm:"size:10;not null;default:'draft';index:idx_status_published" json:"status"` // draft/published
	AIIndexStatus string  `gorm:"size:20;not null;default:'pending'" json:"ai_index_status"`                 // pending/success/failed
	SourceType    string  `gorm:"size:32;not null;default:'manual';index:idx_article_source" json:"source_type"`
	SourceID      *int64  `gorm:"index:idx_article_source" json:"source_id,omitempty"`

	ViewCount    int     `gorm:"default:0" json:"view_count"`
	CommentCount int     `gorm:"default:0" json:"comment_count"`
	LikeCount    int     `gorm:"default:0" json:"like_count"`
	IsTop        bool    `gorm:"default:false" json:"is_top"`
	Popularity   float64 `gorm:"default:0;index:idx_popularity" json:"popularity"` // 活跃度分数

	PublishedAt *time.Time `gorm:"index:idx_status_published" json:"published_at,omitempty"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`

	// 关联（不存储在数据库）
	Category *Category `gorm:"foreignKey:CategoryID" json:"category,omitempty"`
	Author   *User     `gorm:"foreignKey:AuthorID" json:"author,omitempty"`
}

// TableName 指定表名
func (Article) TableName() string {
	return "articles"
}

// IsPublished 判断是否已发布
func (a *Article) IsPublished() bool {
	return a.Status == "published"
}

// IsDraft 判断是否为草稿
func (a *Article) IsDraft() bool {
	return a.Status == "draft"
}

File: frontend/src/types/article.ts
import type { User } from './auth';

export interface Category {
  id: number;
  name: string;
  slug: string;
  description?: string;
  article_count: number;
  sort_order: number;
  created_at: string;
  updated_at: string;
}

export interface Tag {
  id: number;
  name: string;
  slug: string;
  article_count: number;
  created_at: string;
  updated_at: string;
}

export interface Collection {
  id: number;
  name: string;
  slug: string;
  description?: string;
  article_count: number;
  sort_order: number;
  status: 'active' | 'hidden';
  created_at: string;
  updated_at: string;
}

export interface ArticleCollectionMembership {
  collection_id: number;
  name: string;
  slug: string;
  position: number;
}

export interface CollectionNavigationArticle {
  id: number;
  title: string;
  slug: string;
}

export interface ArticleCollectionNavigation {
  collection_id: number;
  collection_name: string;
  collection_slug: string;
  position: number;
  total: number;
  previous?: CollectionNavigationArticle;
  next?: CollectionNavigationArticle;
}

export interface Article {
  id: number;
  title: string;
  slug: string;
  summary: string;
  content: string;
  cover_image?: string;
  status: 'draft' | 'published';
  is_top: boolean;
  ai_index_status: 'pending' | 'success' | 'failed';
  source_type: 'manual' | 'knowledge_document';
  source_id?: number;
  view_count: number;
  like_count: number;
  comment_count: number;
  category_id: number;
  category: Category;
  author_id: number;
  author: User;
  tags?: Tag[];
  collection_id?: number;
  collection_position?: number;
  collection_membership?: ArticleCollectionMembership;
  collection_navigation?: ArticleCollectionNavigation;
  published_at?: string;
  scheduled_publish_at?: string;
  created_at: string;
  updated_at: string;
}

export interface ArticleListItem {
  id: number;
  title: string;
  slug: string;
  summary: string;
  cover_image?: string;
  view_count: number;
  like_count: number;
  comment_count: number;
  status: 'draft' | 'published';
  is_top: boolean;
  ai_index_status: 'pending' | 'success' | 'failed';
  source_type: 'manual' | 'knowledge_document';
  source_id?: number;
  category: Category;
  author: User;
  tags?: Tag[];
  created_at: string;
}

export interface ArticleOrbitCategory {
  id: number;
  name: string;
  slug: string;
}

export interface ArticleOrbitCollection {
  id: number;
  name: string;
  slug: string;
  position: number;
}

export interface ArticleOrbitSemanticPosition {
  x: number;
  y: number;
  z: number;
}

export interface ArticleOrbitSemanticNeighbor {
  article_id: number;
  score: number;
}

export interface ArticleOrbitItem {
  id: number;
  title: string;
  slug: string;
  summary: string;
  cover_image?: string;
  view_count: number;
  comment_count: number;
  is_top: boolean;
  source_type: 'manual' | 'knowledge_document';
  category?: ArticleOrbitCategory;
  collection?: ArticleOrbitCollection;
  semantic_position?: ArticleOrbitSemanticPosition;
  semantic_neighbors?: ArticleOrbitSemanticNeighbor[];
  created_at: string;
  published_at: string;
}

export interface ArticleOrbitResponse {
  data: ArticleOrbitItem[];
  total: number;
}

export interface ArticleInteractionState {
  liked: boolean;
  favorited: boolean;
}

export interface CreateArticleRequest {
  title: string;
  summary: string;
  content: string;
  cover_image?: string;
  category_id: number | undefined;
  status: 'draft' | 'published';
  tag_ids?: number[];
  scheduled_publish_at?: string;
  collection_id?: number;
  collection_position?: number;
}

export interface ArticleSearchResult {
  article: ArticleListItem;
  snippet: string;
  matched_fields: string[];
}
